invenio_users_resources/services/groups/service.py

```diff
@@ -23,23 +32,128 @@
 class GroupsService(RecordService):
     """User groups service."""
 
+    @unit_of_work()
+    def create(self, identity, data, raise_errors=True, uow=None):
+        """Create a new group/role."""
+        self.require_permission(identity, "create")
+        data, errors = self.schema.load(
+            data or {},
+            context={"identity": identity},
+            raise_errors=raise_errors,
+        )
+
+        # Prepare role data for creation
+        name = data.get("name")
+        role_kwargs = {
+            "id": name,
+            "name": name,
+        }
+        if "description" in data:
+            role_kwargs["description"] = data["description"]
+
+        # Create group using API
+        group = GroupAggregate.create(role_kwargs)
```

Comment:
```suggestion
        group = self.record_cls.create(data)
```

Reply:
Also, I think we should be sending to `create` directly `data`. Probably you'll need to "pop" the id because it will come as an empty string (or maybe set it as dump only)